src/main/java/uk/nhs/tis/trainee/notifications/service/EmailService.java

```diff
@@ -122,6 +123,9 @@ private void sendMessage(String traineeId, String recipient, NotificationType no
     String content = templateService.process(templateName, Set.of("content"), templateContext);
 
     MimeMessage mimeMessage = mailSender.createMimeMessage();
+    ObjectId notificationId = ObjectId.get();
```

Comment:
I'm not thrilled with having to generate the ID ourselves, but it seems a better option than the two alternatives:
 1. Store a "sending" state to capture the ID, then update to "sent"
 2. Store the "sent" state to capture the ID before actually sending the email